### PR TITLE
fix(sg): return different error types depending on secret failure

### PR DESCRIPTION
**Pull request by burmudar**
If we failed getting a secret via a tool - we return CommandErr which contains SecretErr
If we failed getting a secret via Google - we return GoogleSecretErr which contains SecretErr

Depending on the error we get while trying to persist Analytics we suggest different fixes the user can try.

Below is how it looks when we get a GoogleSecretErr

Screenshot 2024-07-11 at 11 11 40

## Test plan
Tested locally

### CODE REVIEW

File: dev/sg/internal/secrets/store.go [hunk 1]
```diff
@@ -43,6 +43,32 @@ type Store struct {
...
+type SecretErr struct {
+	Err error
+	Key string
+}
+
+func (se SecretErr) Error() string {
+	return fmt.Sprintf("failed to get secret %q: %v", se.Key, se.Err)
+}
+
+type GoogleSecretErr struct {
```
Comment (jhchabran) [inline dev/sg/internal/secrets/store.go:55]: Some godoc to help understand the context would help 🙏

File: dev/sg/internal/secrets/store.go [hunk 2]
```diff
@@ -130,6 +156,7 @@ func (s *Store) Get(key string, target any) error {
 }
 
 func (s *Store) GetExternal(ctx context.Context, secret ExternalSecret, fallbacks ...FallbackFunc) (string, error) {
+	return "", GoogleSecretErr{}
```
Comment (jhchabran) [inline dev/sg/internal/secrets/store.go:159]: left over 😊

File: dev/sg/internal/secrets/store.go [hunk 3]
```diff
@@ -43,6 +43,32 @@ type Store struct {
...
+func (gse GoogleSecretErr) Error() string {
+	return fmt.Sprintf("google(%s): %s", gse.Project, gse.SecretErr.Error())
+}
+
+type CommandErr struct {
+	SecretErr
+}
+
+func (ce CommandErr) Error() string {
+	return fmt.Sprintf("command error - %v", ce.SecretErr.Error())
```
Comment (jhchabran) [inline dev/sg/internal/secrets/store.go:69]:
```suggestion
	return fmt.Sprintf("command error: %v", ce.SecretErr.Error())
```

**Review by jhchabran — CHANGES_REQUESTED** [review]
Nice! Good call!

Requested changes, because left-over + need some godoc 🙏

File: dev/sg/internal/secrets/store.go (hunk 2 above)
Comment (burmudar) [reply dev/sg/internal/secrets/store.go]: Whoops thanks! Removed 🙏🏼

File: dev/sg/internal/secrets/store.go (hunk 1 above)
Comment (burmudar) [reply dev/sg/internal/secrets/store.go:61]: Good call! I've added it now

**Review by jhchabran — APPROVED** [review]

**Pull request merged**